src/components/vehicleReference/index.tsx

```diff
@@ -1,33 +1,33 @@
 import React, { ComponentProps, FC, ReactNode } from 'react';
 
-import { Box, chakra, useMultiStyleConfig } from '@chakra-ui/react';
+import { chakra, useSlotRecipe } from '@chakra-ui/react';
 
-import Stack from '../stack';
+import { AspectRatio, Box, Grid, MissingImage, Stack } from 'src';
 
-import MissingImage from '../missingImage';
-import Grid from '../grid';
-import AspectRatio from '../aspectRatio';
-
-interface Props {
+export type VehicleReferenceProps = {
   image?: ReactNode;
   vehicleTitle: string;
   price?: string | null;
   sellerName?: string | null;
   sellerAddress?: string | null;
   callToAction?: ReactNode;
   templateColumns?: ComponentProps<typeof Grid>['templateColumns'];
-}
+};
 
-const VehicleReference: FC<Props> = ({
-  image,
-  vehicleTitle,
-  price,
-  sellerName,
-  sellerAddress,
-  callToAction,
-  templateColumns = { base: 'auto 1fr', md: '1fr' },
-}) => {
-  const styles = useMultiStyleConfig(`VehicleReference`);
+export const VehicleReference: FC<VehicleReferenceProps> = (props) => {
+  const recipe = useSlotRecipe({ key: 'VehicleReference' });
+  const [recipeProps, componentProps] = recipe.splitVariantProps(props);
```

Comment:
Step 2.
`splitVariantProps` splits the properties to the ones that recipe variations would expect (eg. `size` or `variant` - they need to be defined in the recipe itself) and the ones the component expects.